### PR TITLE
swiftformat: add linux plartform, 0.47.10 -> 0.54.3

### PR DESCRIPTION
**Pull request by ilaumjd**
## Description of changes

- Adding linux platform
- Updating to the latest version

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/swiftformat/default.nix [hunk 1]
```diff
@@ -1,45 +1,46 @@
...
 
-  sandboxProfile = ''
-    (allow file-read* file-write* process-exec mach-lookup)
-    ; block homebrew dependencies
-    (deny file-read* file-write* process-exec mach-lookup (subpath "/usr/local") (with no-log))
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 ${srcName} $out/bin/swiftformat
+    runHook postInstall
   '';
```
Comment (bdesham) [inline pkgs/development/tools/swiftformat/default.nix:36]:
```suggestion
  '';
  
  passthru.tests.version = testers.testVersion { package = swiftformat; };
```

File: pkgs/development/tools/swiftformat/default.nix [hunk 2]
```diff
@@ -1,45 +1,46 @@
...
-
-# This derivation is impure: it relies on an Xcode toolchain being installed
-# and available in the expected place. The values of sandboxProfile and
-# hydraPlatforms are copied pretty directly from the MacVim derivation, which
-# is also impure.
-
+{
+  stdenv,
+  lib,
+  fetchurl,
```
Comment (bdesham) [inline pkgs/development/tools/swiftformat/default.nix:4]:
```suggestion
  fetchurl,
  swiftformat,
  testers,
```

**Review by bdesham — APPROVED** [review]
These changes look good, thank you for doing this! My only suggestion is to add a simple sanity check that uses `testers.testVersion` to run `swiftformat --version` and verify that the executable runs as expected.

**Review by bdesham — APPROVED** [review]

**Comment by ilaumjd** [comment]
Why do it got exec format error? Is ofborg using NixOS and cannot execute the binary? Because i don't use NixOS (i'm on Fedora) and build is fine on me.
https://github.com/NixOS/nixpkgs/pull/338533/checks?check_run_id=29569058802

**Comment by bdesham** [comment]
It looks like the Linux binary offered by SwiftFormat is x86-64, but ofborg is trying to run the test on an aarch64 system. I think this derivation needs to be limited to x86 Linux and aarch64 macOS.

**Comment by ilaumjd** [comment]
the error is also in x86_64-linux https://github.com/NixOS/nixpkgs/pull/338533/checks?check_run_id=29569087192

**Comment by bdesham** [comment]
Ah. The error for x86_64-linux is different:

    {{storeDir}}/bin/swiftformat: cannot execute: required file not found

When this derivation is run under Linux, the binary is named `swiftformat_linux`, not `swiftformat`. I think we should probably rename `swiftformat_linux` to `swiftformat` after unzipping when the user is on Linux.

**Comment by bdesham** [comment]
The necessary change is probably something like replacing

```nix
  installPhase = ''
    runHook preInstall
    install -Dm755 ${srcName} $out/bin/swiftformat
    runHook postInstall
  '';
```

with

```nix
  installPhase = if stdenv.isDarwin then ''
      install -Dm755 ${srcName} $out/bin/swiftformat
    '' else ''
      install -Dm755 ${srcName}_linux $out/bin/swiftformat
    '';
```

I don’t think it’s necessary to run the `preInstall` and `postInstall` hooks, since you haven’t defined any 🙂

**Comment by ilaumjd** [comment]
Sorry, been a long time since i take a look at this.
Yeah, i will update to remove aarch64-linux and the hooks, also with update to the latest version.

```
 install -Dm755 ${srcName}_linux $out/bin/swiftformat
```
As for this, `srcName` already contains `_linux` so this should not be necessary. I was wondering why is file not found on x86_64-linux, i'm using this for my local testing and it works.

**Comment by ilaumjd** [comment]
I don't understand what is happening here, iirc on `aarch64-linux` it was a binary error, which means the binary file are there right, but why file is not found on `x86_64-linux`? Can anyone help?

**Pull request closed without merging**